cartoframes/context.py

```diff
@@ -1021,62 +1138,96 @@ def data_augment(self, table_name, metadata):
+                       count(*)::int AS cnt
+                  FROM {table_name}
+              )
+              SELECT *
+                FROM json_to_recordset(
+                    (SELECT OBS_GetMeta(
+                        envelope.env,
+                        ('{meta}')::json,
+                        10, 1, envelope.cnt
+                    ) AS meta
+                FROM envelope
+                GROUP BY env, cnt)) as data(
+                    denom_aggregate text, denom_colname text,
+                    denom_description text, denom_geomref_colname text,
+                    denom_id text, denom_name text, denom_reltype text,
+                    denom_t_description text, denom_tablename text,
+                    denom_type text, geom_colname text, geom_description text,
+                    geom_geomref_colname text, geom_id text, geom_name text,
+                    geom_t_description text, geom_tablename text,
+                    geom_timespan text, geom_type text, id numeric,
+                    max_score_rank text, max_timespan_rank text,
+                    normalization text, num_geoms numeric,numer_aggregate text,
+                    numer_colname text, numer_description text,
+                    numer_geomref_colname text, numer_id text,
+                    numer_name text, numer_t_description text,
+                    numer_tablename text, numer_timespan text,
+                    numer_type text, score numeric, score_rank numeric,
+                    score_rownum numeric, suggested_name text,
+                    target_area text, target_geoms text, timespan_rank numeric,
+                    timespan_rownum numeric)
+            '''.format(
+                    table_name=table_name,
+                    meta=json.dumps(metadata).replace('\'', '\'\''))
+            resp = self.sql_client.send(query)
+            _meta = pd.DataFrame(resp['rows'])
+
+        # TODO: add OBS_GetMetadataValidation here?
+        tablecols = self.sql_client.send('''
+            SELECT * FROM {table_name} LIMIT 0
+            '''.format(table_name=table_name),
+            **DEFAULT_SQL_ARGS)['fields'].keys()
+
+        if set(tablecols) & set(_meta['suggested_name']):
+            commoncols = set(tablecols) & set(_meta['suggested_name'])
+            raise NameError('Column name collision for column(s): {cols}. '
+                            'Rename table column(s) to resolve.'.format(
+                                cols=', '.join(commoncols)))
+
+        cols = ', '.join(
+                '(data->{n}->>\'value\')::{pgtype} AS {col}'.format(
+                    n=row[0],
+                    pgtype=row[1]['numer_type'],
+                    col=row[1]['suggested_name'])
+                for row in _meta.iterrows()
+            )
+        query = '''
+            SELECT t.*, {cols}
+              FROM OBS_GetData(
+                   (SELECT array_agg((the_geom, cartodb_id)::geomval)
```

Comment:
@javitonino, is it possible to do this by `geom_ref` instead of a geometry similar to how `OBS_GetMeasureById` vs. `OBS_GetMeasure`?

Reply:
Yes. You can pass a text array to GetData as the first parameter, containing geoids, instead of the geomvals. Can I ask why? (just curious)

Reply:
If a user has a dataset with geoids, then they can get raw measures. Oftentimes people want just the raw measures without having any calculations (like spatial interpolation) performed on them.

Reply:
I somehow missed the `geom_ref text[]` version of the function in the docs: https://carto.com/docs/carto-engine/data/measures-functions/#obs_getdataids-arraytext-metadata-json Sorry for the noise!